### ISSUE TITLE
Developer docs: setting up a local development environment

### ISSUE THREAD

**Issue by bugfolder** [backdrop-ops/docs.backdropcms.org#189]
We're going to start a page on the docs site on (probably multiple) ways to set up a development environment as a resource for new contributors. It will be a sub-page of https://docs.backdropcms.org/documentation/developing-for-backdrop. Please post your descriptions in the comments on this issue, and I'll synthesize them into the docs page.

**Comment by stpaultim** [comment]
Yes, I think we should start with a simple description of what it means to set up a local dev environment and why you need to do it. Example: "A local dev environment simulates the server software that is necessary to run a website, while this software may be available on your computer, it may require some custom configuration before you can use it to run Backdrop CMS on your desktop or laptop." 

I think it is probably fine to list the known options and then link off-site (as available) to other documentation.

I'm a Lando user on Linux (occasionally DDev). https://docs.lando.dev/backdrop/
DDev = https://backdropcms.org/news/getting-started-with-backdrop-and-ddev-local

**Comment by bugfolder** [comment]
Here's where contributions here will end up. https://docs.backdropcms.org/documentation/setting-up-a-local-development-environment

**Comment by oadaeh** [comment]
A) That is not a Backdrop specific question/problem.
B) There are many ways to do it and many ways to have problems doing it - we're not going to cover all the bases.
C) There are also lots of howtos/tutorials/etc already on the Internet for how to set up a development environment.
Why not make use of what is already available?

**Comment by bugfolder** [comment]
@oadaeh, we can link to offsite tutorials (and that's a good suggestion, to lead with those links). But having a Backdrop-specific instruction would be helpful (especially to my former self).

**Comment by stpaultim** [comment]
> Why not make use of what is already available?

I agree. I think the point it is to start with an explanation of why you might need to do this when working with Backdrop CMS, which seems obvious to us, but might not be to a complete beginner and to list the options. 

I do not think we should try to write new documentation for each specific method. But, we need a good starting place for folks just getting started and a list of the various options.

**Comment by bugfolder** [comment]
@**klonos** said (on Zulip):

> if you do a MAMP one, I can do a Docker/Lando one.

Deal. How about if you start with the Docker/Lando version (post as a comment here); I'd like to see your presentation to use as a model for the MAMP version. I've already added some intro to the docs page.

**Comment by klonos** [comment]
@bugfolder I think I'll create it as a documentation article/page (draft - unpublished). I'll post a link here once done.

**Comment by yorkshire-pudding** [comment]
I may try to do a Windows one.  I'm currently using Lando on Ubuntu through Windows Subsystem for Linux (WSL) on Windows 10 with Visual Studio Code as my code editor.  As most documentation assumes you're on a Mac, it might be useful for some.  I don't have time at the moment, but hopefully will soon.

**Comment by klonos** [comment]
Thank you @yorkshire-pudding for offering to do that 🙏🏼  ...same here re lack of time, but I will do my best to get the Mac/Lando one done as soon as possible.

**Comment by bugfolder** [comment]
[Notes from @stpaultim](https://github.com/backdrop/backdrop-issues/issues/5443#issuecomment-1063693194) to use as source for sub-pages:

>Here are three possible tools you can use:
Lando = [docs.lando.dev/backdrop](https://docs.lando.dev/backdrop/)
DDev = [backdropcms.org/news/getting-started-with-backdrop-and-ddev-local](https://backdropcms.org/news/getting-started-with-backdrop-and-ddev-local)
MAMP = https://www.jenlampton.com/blog/using-mamp-local-backdrop-cms-development
